### PR TITLE
docs(dashboard-grid): add initial docs to cover DashboardGrid

### LINKED ISSUE #3256: [Dashboard Grid] documentation seems to be missing from the story book
### What package is this for?

- [ X] React
- [ ] Angular

### Describe the bug

Documentation for this component seems to be missing from the stroybook

### To Reproduce

Navigate to the docs section of the storybook for this componet

### PR DESCRIPTION
**Pull request by kevinsperrine**
Closes #3256 

**Summary**

- add documentation to cover DashboardGrid usage

**Change List (commits, features, bugs, etc)**

- new docs

**Acceptance Test (how to verify the PR)**

- proofread docs
- check links work in [storybook docs](https://deploy-preview-3273--carbon-addons-iot-react.netlify.app/?path=/docs/1-watson-iot-dashboard-grid--dashboard-default-layouts)

**Regression Test (how to make sure this PR doesn't break old functionality)**

- n/a

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

File: packages/react/src/components/Dashboard/DashboardGrid.mdx [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+      }}
+      content="My utilization chart"
+    />,
+  ]}
+</DashboardGrid>
+```
+
+## Resizable cards
+
+Mby passing `isResizable` to the cards in a DashboardGrid, the cards become resizable by dragging and
```
Comment (davidicus) [inline packages/react/src/components/Dashboard/DashboardGrid.mdx:152]: Typo
```suggestion
By passing `isResizable` to the cards in a DashboardGrid, the cards become resizable by dragging and
```

**Comment by kevinsperrine** [comment]
> This is great! Thanks for putting this together. My only question would be, does it make sense to add a line about what you would do with `onCardSizeChange` and `onLayoutChange`.

Yeah, I added a couple lines related to that.

File: packages/react/src/components/Dashboard/DashboardGrid.mdx [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+| layouts.xs[].h  | number          |         | The height of this card at the xs breakpoint      |
+
+## Source Code
+
+[Source code](https://github.com/carbon-design-system/carbon-addons-iot-react/tree/next/packages/react/src/components/DashboardGrid)
+
+## Feedback
+
+Help us improve this component by providing feedback, asking questions on Slack, or updating this file on
+[GitHub](https://github.com/carbon-design-system/carbon-addons-iot-react/tree/next/packages/react/src/components/DashboardGrid/DashboardGrid.mdx).
```
Comment (bjornalm) [inline packages/react/src/components/Dashboard/DashboardGrid.mdx:273]: Should be https://github.com/carbon-design-system/carbon-addons-iot-react/blob/next/packages/react/src/components/Dashboard/DashboardGrid.mdx

File: packages/react/src/components/Dashboard/DashboardGrid.mdx [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+| layouts.xs      | arrayOf(object) |         |                                                   |
+| layouts.xs[].i  | string          |         | the id of this element                            |
+| layouts.xs[].x  | number          |         | The x position of this card at the xs breakpoint  |
+| layouts.xs[].y  | number          |         | The y position of this card at the xs breakpoint  |
+| layouts.xs[].w  | number          |         | The width of this card at the xs breakpoint       |
+| layouts.xs[].h  | number          |         | The height of this card at the xs breakpoint      |
+
+## Source Code
+
+[Source code](https://github.com/carbon-design-system/carbon-addons-iot-react/tree/next/packages/react/src/components/DashboardGrid)
```
Comment (bjornalm) [inline packages/react/src/components/Dashboard/DashboardGrid.mdx:268]: Should be https://github.com/carbon-design-system/carbon-addons-iot-react/blob/next/packages/react/src/components/Dashboard/DashboardGrid.jsx

File: packages/react/src/components/Dashboard/DashboardGrid.mdx [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+      }}
+      content="My Facility Metrics"
+    />,
+  ]}
+</DashboardGrid>;
+```
+
+## Custom layout
+
+Pass a custom layout to the dashboard grid to assign positions and sizes to the various to the cards.
```
Comment (bjornalm) [inline packages/react/src/components/Dashboard/DashboardGrid.mdx:39]: spelling: to the various cards

File: packages/react/src/components/Dashboard/DashboardGrid.mdx [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+      content="My utilization chart"
+    />,
+  ]}
+</DashboardGrid>
+```
+
+## Editable layout
+
+You can drag and drop the cards around. The `onLayoutChange` handler will be triggered each time you
+drop a card in a new location. The card array contains objects representing each card in the layout.
```
Comment (bjornalm) [inline packages/react/src/components/Dashboard/DashboardGrid.mdx:104]: ...card in the layout, similar to the layout...

File: packages/react/src/components/Dashboard/DashboardGrid.mdx [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+Similar to the layout prop above. It's easiest to inspect the output in the Actions tab of the
+[is editable story](/?path=/story/1-watson-iot-dashboard-grid--dashboard-is-editable) to see the
```
Comment (bjornalm) [inline packages/react/src/components/Dashboard/DashboardGrid.mdx:106]: Link is not working, remove the section `/?path=` 
Story links must start with `/story/`

File: packages/react/src/components/Dashboard/DashboardGrid.mdx [hunk 7]
```diff
@@ -0,0 +1,273 @@
...
+```
+
+## Resizable cards
+
+By passing `isResizable` to the cards in a DashboardGrid, the cards become resizable by dragging and
+the card size prop is automatically updated to match the new size during the drag process. This triggers
+the `onCardSizeChange` callback which is passed two parameters. The first is an object giving details
+about the card (its `id` and its `size`), and the second is an object containing information about
+the DashboardGrid and its layout. Again, it's easiest to view the Actions panel in the [resizable card
+story](/?path=/story/1-watson-iot-dashboard-grid--dashboard-resizable-card) to see callback in action.
```
Comment (bjornalm) [inline packages/react/src/components/Dashboard/DashboardGrid.mdx:157]: Remove section `/?path=`

File: packages/react/src/components/Dashboard/DashboardGrid.mdx [hunk 8]
```diff
@@ -0,0 +1,273 @@
...
+The `onCardSizeChange` is triggered instantly as you drag and drop and the `onResizeStop` callback is
+only triggered once you've stopped resizing the card. The `onResizeStop` or `onCardSizeChange` callbacks
+can be used to save the layout state, so that it can be recreated later.
```
Comment (bjornalm) [inline packages/react/src/components/Dashboard/DashboardGrid.mdx:160]: re-created?

**Review by bjornalm — CHANGES_REQUESTED** [review]
Nice work, just some spelling mistakes and links that need fixing

**Review by bjornalm — APPROVED** [review]

**Pull request merged**